Cargo.toml

```diff
@@ -18,6 +18,7 @@ maintenance = { status = "actively-developed" }
 [dependencies]
 async-trait = "0.1.77"
 lazy_static = "1.4.0"
+mockall = { version = "0.12.1", optional = true }
```

Comment:
Any reason to add this as runtime dependency? If someone wants to mock `FeatureProvider` as an SDK user, then they can use `mock! { ... }` from `mockall`.

Reply:
> Any reason to add this as runtime dependency? If someone wants to mock `FeatureProvider` as an SDK user, then they can use `mock! { ... }` from `mockall`.

That is a very good point. I have been thinking about it for a while...Yes, we can add it as a dev dependency. But in that case, the consumer is not able to use `MockFeatureProvider` any more, unless they mock the trait by themselves by repeating the trait definition.

Another way to implement it, which is the way I am currently taking, is to add an optional feature `mockall` that is enabled by default. The user can exclude this feature if not needed.

I am open to further discussions. Any idea/proposal/opinion is welcomed! :-)

Reply:
gotcha, maybe instead of calling it `mockall`, maybe something like `test-util`, like in `tokio`, then we can also add other convenience functionality around testing. I don't think this will be the last one 😄

Reply:
Advice taken! Really appreciate it. :-D